### PR TITLE
refactor pg snippets

### LINKED ISSUE #668: Change pg examples from callbacks to Promises
**Which module(s) and week(s) does this change affect?**
Module(s): SQL
Week(s): 2/3
 
**What is the work that needs to be done?**

Currently the node pg examples make use of an old continuation passing style where callback functions are passed as arguments to methods from the pg API. We just need to refactor these snippets so they use Promises instead.
E.g:
```js
  const query =
  "INSERT INTO customers (name, email, phone, address, city, postcode, country) " +
    "VALUES ($1, $2, $3, $4, $5, $6, $7) " +
    "RETURNING id";                         // Note the new clause
  db.query(query, [newName, newEmail, ..., newCountry],
    (err, result) => {                      // note the addition of result parameter
      if (err == undefined) {
        const newId = result.rows[0].id;
        console.log(`New Customer id = ${newId}`);
        res.status(200).json({lastId: newId});
      } else {
        res.status(500).json({error: err});
      };
    });
```

**Why is this work important to do?**

We want to favour Promises over callbacks wherever possible.

**Additional context**

**Who might need to know about this change?**

### PR DESCRIPTION
**Pull request by Dedekind561**
## What does this change?

Module: SQL 
Week(s): 1,2,3

Fixes: #668 

## Checklist

- [x] I have read the [contributing guidelines](https://syllabus.codeyourfuture.io/contributing/overview)
- [x] I have checked my spelling and grammar with an [automated tool](https://www.grammarly.com/grammar-check)
- [x] I have previewed my changes to check the [markdown renders](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) as I intend
- [x] I have run my code to check it works
- [x] My changes follow our [Style Guide](https://syllabus.codeyourfuture.io/guides/code-style-guide)

## Description

- All snippets using callbacks refactored to use Promises
- String concatenation changed over to template literals
- `var` -> `const`
- Nested promise changes removed
- Markdown updated wherever necessary to reflect updates to code snippets

### CODE REVIEW

File: docs/db/week-3/lesson.md [hunk 1]
```diff
@@ -344,45 +337,35 @@ app.post("/customers", function (req, res) {
...
-            res.send("Customer created.");
-          }
-      }
-    });
-});
+db.query("SELECT 1 FROM customers WHERE email=$1", [newEmail])
+  .then(function(result) {
+    if (result.rowCount > 0) {      // note the use of result.rowCount
+      return Promise.reject({error: 'Customer already exists'})
+    } else {
```
Comment (Dedekind561) [inline docs/db/week-3/lesson.md:350]: This is a naive implementation. Better to add a unique constraint on the `email` column: however I left this in as it is a good opportunity for a refactor. Worthwhile to come up with an exercise on this.

File: docs/db/week-3/lesson.md [hunk 2]
```diff
@@ -58,9 +58,13 @@ const db = new Pool({
 });
 
 app.get("/customers", function (req, res) {
-  db.query("SELECT * FROM customers", function(err, result) => {
-    res.status(200).json({customers: result.rows})
-  })
+  db.query("SELECT * FROM customers")
+    .then((result) => {
+      res.status(200).json({ customers: result.rows });
```
Comment (jackhclee) [inline docs/db/week-3/lesson.md:63]: Refer to week3 line #713, shall we return an array or an object with `customers` properties holding the array?

File: docs/db/week-3/lesson.md [hunk 3]
```diff
@@ -101,21 +105,21 @@ Here the endpoint includes `:id`, which identifies an extra parameter in the URL
...
-    // TODO - more code here...
-  }
-);
+db.query("SELECT * FROM customers WHERE id = $1", [custId])
+  .then(function (result) {
+    console.log(result);
+  })
+  .catch(function (err) {
+    console.log(err);
+  });
```
Comment (jackhclee) [inline docs/db/week-3/lesson.md]: Shall we use arrow function here?

Comment (SallyMcGrath) [reply docs/db/week-3/lesson.md]: Hey @jackhclee 

Could you take a look through https://github.com/CodeYourFuture/Module-Databases when you have a sec?

I'm debating letting L9 use the new coursework as well as the new DB module, but I really need some eyes on it first.

Comment (Dedekind561) [reply docs/db/week-3/lesson.md]: @SallyMcGrath I agree re LDN9 moving to Github projects for the database module too
I had a cursory look at the Database Module and those issues made sense to me

Comment (Dedekind561) [reply docs/db/week-3/lesson.md]: I'm going to clone it down now and check it makes sense

File: docs/db/week-3/lesson.md (hunk 2 above)
Comment (Dedekind561) [reply docs/db/week-3/lesson.md:63]: If it's an API response it should be an object of the form `{ resourceName : x }`. Is this what you mean?

**Comment by Dedekind561** [comment]
@jackhclee I'm going to convert all callback functions in the `then` and `catch` blocks to arrow functions

File: docs/db/week-3/lesson.md (hunk 3 above)
Comment (SallyMcGrath) [reply docs/db/week-3/lesson.md]: I think PD might have a problem with this. It's unclear. We might find out tonight.

Comment (SallyMcGrath) [reply docs/db/week-3/lesson.md]: Daniel has reviewed, but I forgot to ask Karen about L9 (and in fact this conversation).

Comment (kfklein15) [reply docs/db/week-3/lesson.md]: Why could there be a problem for PD with this, @SallyMcGrath ?

Comment (SallyMcGrath) [reply docs/db/week-3/lesson.md]: Dumping Google Classroom and using boards? Last time we spoke we agreed to do a trial of JS3 in L10/GL6 only, but this conversation is about Databases for L9 @kfklein15

Comment (kfklein15) [reply docs/db/week-3/lesson.md]: I am not sure I understand what the problem for PD would be. Sorry, but are you suggesting we would have to change this too into the board? If yes, we need to have a clear timeline.

Comment (SallyMcGrath) [reply docs/db/week-3/lesson.md]: Yes, this conversation has me and Mitch discussing L9 moving to the new coursework for the DB module. We have a completely new DB module for L9 and it has no coursework set for it on the syllabus. We have written new coursework, but it's in the [Module repo](https://github.com/CodeYourFuture/Module-Databases). It's ready to go, but this conversation is me saying that PD might have a problem with us using it, in which case we will have to use the old coursework.

**Pull request merged**